File: src/main/java/com/fml/common/auth/PasswordHasher.java
package com.fml.common.auth;

/** Salted SHA-256 password hashing used by every FML service. */
public final class PasswordHasher {
    private PasswordHasher() {
    }

    public static String hash(String salt, String password) {
        return Sha256Hex.of(salt + password);
    }
}

File: src/Fml.Common/Auth/PasswordHasher.cs
using System.Security.Cryptography;
using System.Text;

namespace FML.Common.Auth;

/// <summary>Salted SHA-256 password hashing used by every FML service.</summary>
public static class PasswordHasher
{
    public static string Hash(string salt, string password) =>
        Convert.ToHexString(SHA256.HashData(Encoding.UTF8.GetBytes(salt + password)));
}
